### PR TITLE
fixes #7592 test(nimbus): Added end-to-end test for telemetry verification during enrollment and unenrollment

### LINKED ISSUE #7592: Add check for telemetry into integration tests
To increase the exhaustiveness of our integration tests, we should add a check for telemetry events. The 2 major events we are looking for are enrollment and unenrollment. We should also check for a paused event.

Using [pytest-pings](https://github.com/hackebrot/pytest-pings) we can collect the ping data sent from firefox and validate it.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2681)

### PR DESCRIPTION
**Pull request by jrbenny35**
Because
* Making sure telemetry is sent when a client enrolls and unenrolls is very important. We did not have automated tests for this though

This Commit
* Adds automated checks for telemetry enroll/unenroll events.

Since this test can take a varying amount of time to complete, I utilized pytest-xdist to group tests so that this test can be run immediately, to save time.
I also had to write a ping server to collect and verify the pings, that is included in this PR.

### CODE REVIEW

**Comment by yashika-khurana** [comment]
@jrbenny35 oh I see one check is failing :(

**Comment by jrbenny35** [comment]
> @jrbenny35 oh I see one check is failing :(

Yeah, seems another test is failing. Not sure why exactly

**Comment by yashika-khurana** [comment]
@jrbenny35 it seems like error in screenshot test case and not able to find particular file

**Comment by jrbenny35** [comment]
> @jrbenny35 it seems like error in screenshot test case and not able to find particular file

Yup, it was a pref I had set. I moved all the prefs regarding the telemetry tests into the file with the test

**Comment by jrbenny35** [comment]
Just making sure everything works on release now

File: app/tests/integration/nimbus/conftest.py [hunk 1]
```diff
@@ -235,4 +237,100 @@ def _create_mobile_experiment(name, app, languages, targeting):
...
+                            id
+                        }
+                    }
+                    """,
+        }
+
+        response = requests.post(
+            "https://nginx/api/v5/graphql", json=get_id_query, verify=False
+        )
+        print(response.text)
```
Comment (yashika-khurana) [inline app/tests/integration/nimbus/conftest.py:269]: oops print statement!

File: app/tests/integration/nimbus/conftest.py [hunk 2]
```diff
@@ -235,4 +237,100 @@ def _create_mobile_experiment(name, app, languages, targeting):
...
+                    "populationPercent": data.get("population_percent"),
+                    "totalEnrolledClients": data.get("total_enrolled_clients"),
+                }
+            },
+            "query": "mutation updateExperiment($input: ExperimentInput!) \
+                {\n updateExperiment(input: $input) \
+                    {\n message\n __typename\n }\n}\n",
+        }
+        response = requests.post("https://nginx/api/v5/graphql", json=query, verify=False)
+        print(response.text)
```
Comment (yashika-khurana) [inline app/tests/integration/nimbus/conftest.py:298]: same here!

**Review by yashika-khurana — CHANGES_REQUESTED** [review]
LGTM!, just minor comment, great job @jrbenny35 👏 👏 🎉

**Comment by yashika-khurana** [comment]
@jrbenny35 it looks good to me, I am just wondering if we can write a test case where enrollment shouldn't happen? so that we can make sure enrollment doesn't happen in those cases where it shouldn't. Just a thought 🤔

**Comment by jrbenny35** [comment]
> @jrbenny35 it looks good to me, I am just wondering if we can write a test case where enrollment shouldn't happen? so that we can make sure enrollment doesn't happen in those cases where it shouldn't. Just a thought 🤔

Yeah, that wouldn't be hard. Just check for the ```enrollFailed``` event and the reason being a JEXL mismatch or something. I can file that on JIRA

**Comment by yashika-khurana** [comment]
@jrbenny35 perfect, lets merge this and we can add that next. Thanks @jrbenny35 its awesome, great job 🚀

**Review by yashika-khurana — APPROVED** [review]

**Pull request merged**